File: src/problems/Problem10.java
package problems;

import java.util.List;
import util.SieveOfEratosthenes;

/**
 *
 * @author ataylor
 */
public class Problem10 {
    
    private SieveOfEratosthenes sieve;
    private int size = 2000000; // 2 million
    
    public Problem10() {
        sieve = new SieveOfEratosthenes(size);
        sieve.generate(true);
    }
    
    public void solve() {
        List<Integer> primes = sieve.getPrimes();
        
        long sum = 0;
        for (int p : primes)
            sum += p;
        
        System.out.println(sum);
    }
    
    public static void main(String[] args) {
       new Problem10().solve();
    }
}

File: python/problem10.py
#!/usr/bin/python

def sieve_of_eratosthenes(n):
	sieve = {}

	for x in range(2, n):
		sieve[x] = True

	for y in range(2, n):
		if sieve[y]:
			z = y 
			while z + y < n:
				z = z + y
				sieve[z] = False

	return sieve

sum = 0
n = 2000000
sieve = sieve_of_eratosthenes(n)
print sieve

for x in range(2, n):
	if sieve[x]:
		sum += x

print sum
